functions/src/index.ts

```diff
@@ -34,14 +34,17 @@ async function getData(gymName: string, startDate: Date, endDate: Date, offset:
 
     // dates
     const dates = []; // list of Date objects from startDate to endDate
-    for (const i = new Date(startDate.getTime()); i < endDate; i.setDate(i.getDate() + 1)) {
-        dates.push(new Date(i.getTime() - offset * 60000)); // local time
+    for (const i = new Date(startDate); i < endDate; i.setTime(i.getTime() + 24 * 60 * 60 * 1000)) {
+        dates.push(new Date(i)); // UTC
     }
-    const dateHeader = dates.map(d => d.toLocaleString("en-US", { // local time string
-        weekday: 'short',
-        month: '2-digit',
-        day: '2-digit'
-    }));
+    const dateHeader = dates.map(d => {
+        return d.toLocaleString("en-US", { // UTC date string
```

Comment:
this isn't a big deal, but I think you could get rid of the brackets and `return` statement:

```typescript
... = dates.map(d =>
  d.toLocaleString( 
    ...
  )
);
```

Reply:
haha I was experimenting and forgot to change it back, good catch